libraries/Microsoft.Bot.Builder.Azure/AzureBlobStorage.cs

```diff
@@ -79,14 +76,14 @@ public async Task DeleteAsync(string[] keys, CancellationToken cancellationToken
                 throw new ArgumentNullException(nameof(keys));
             }
 
-            var blobContainer = await GetBlobContainer().ConfigureAwait(false);
-            await Task.WhenAll(
-                keys.Select(key =>
-                {
-                    var blobName = GetBlobName(key);
-                    var blobReference = blobContainer.GetBlobReference(blobName);
-                    return blobReference.DeleteIfExistsAsync();
-                })).ConfigureAwait(false);
+            var blobClient = _storageAccount.CreateCloudBlobClient();
+            var blobContainer = blobClient.GetContainerReference(_containerName);
+            foreach (var key in keys)
+            {
+                var blobName = GetBlobName(key);
+                var blobReference = blobContainer.GetBlobReference(blobName);
+                await blobReference.DeleteIfExistsAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
+1 to use of Cancellation Token here. 

Should you be checking between iterations to see if it was cancelled, then bail out? Or does that happen automatically through Exception handling?